source/client-side-encryption/client-side-encryption.rst

```diff
@@ -385,8 +385,35 @@ to authenticate.
       key: byte[96] or String // The master key used to encrypt/decrypt data keys. May be passed as a base64 encoded string.
    }
 
+   kmip: {
+      endpoint: String
+   }
+
 See `Why are extraOptions and kmsProviders maps?`_
 
+kmip provider TLS options
+`````````````````````````
+
+Drivers MUST provide TLS options to configure TLS connections for the "kmip"
+provider.
+
+The TLS options MUST be equivalent to the existing TLS options for MongoDB
+server TLS connections.
+
+Drivers SHOULD provide API that is consistent with configuring TLS options for
+MongoDB server TLS connections. New API to support the options MUST be
+independent of the KMS provider to permit future extension. The following is an
+example:
+
+.. code:: typescript
+
+   class AutoEncryptionOpts {
+      // setTLSOptions throws an exception if kmsProvider is not "kmip".
+      setTLSOptions (kmsProvider String, tlsOptions TLSOptions)
+   }
+
+See `Why does the KMIP provider require TLS configuration?`_
```

Comment:
Whether this is feasible for all drivers is an open question.

The [URI options](https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst#list-of-specified-options) alludes to driver limitations on TLS options:

> If the driver language/runtime does not allow non-global configuration of TLS

See [this spreadsheet](https://docs.google.com/spreadsheets/d/1dPHcHIz182p-Ygm1IQF7oXn7aGId115sIAlcSl-87Wo/edit#gid=0 ) for driver support of TLS options on the `MongoClient`.

If my understanding is correct, all drivers support TLS configuration of an equivalent to the `tlsCertificateKeyFile` option on a `MongoClient` (i.e. not only at the global level). C# does not support `tlsCAFile` equivalent. I think that is OK since a user could install the CA for KMIP system-wide.

Reply:
This seems fine for Java driver.  The configuration would just be an optional instance of `javax.net.ssl.SSLContext`.  
Is it a good assumption that disabling TLS should not be supported?  Same goes for invalidHostNameAllowed.